pkg/config/config.go

```diff
@@ -65,11 +45,32 @@ func LoadConfig(
 			filepath.Join(usr.HomeDir,
 				".kube",
 				"config"))
-		if c, err := clientcmd.BuildConfigFromFlags(url, filepath.Join(usr.HomeDir, ".kube", "config")); err == nil {
-			return c, nil
-		}
+		return configFromFile(url, filepath.Join(usr.HomeDir, ".kube", "config"), context)
 	}
 
 	return nil, fmt.Errorf("could not create a valid kubeconfig")
 
 }
+
+func configFromFile(url, kubeconfig, context string) (*rest.Config, error) {
+	if context == "" {
+		// klog.V(5).Infof("clientcmd.BuildConfigFromFlags with %s and %s", url, kubeconfig)
+		// Retreive the config for the current context
+		if url == "" {
+			config, err := clientcmd.LoadFromFile(kubeconfig)
+			if err != nil {
+				return nil, err
+			}
+			return clientcmd.NewDefaultClientConfig(
+				*config,
+				&clientcmd.ConfigOverrides{}).ClientConfig()
+		}
+		return clientcmd.BuildConfigFromFlags(url, kubeconfig)
+	} else {
```

Comment:
no need for else.. just return statement

Reply:
Done